presto-mysql/src/main/java/com/facebook/presto/plugin/mysql/MySqlClient.java

```diff
@@ -111,10 +115,24 @@ protected SchemaTableName getSchemaTableName(ResultSet resultSet)
     @Override
     protected String toSqlType(Type type)
     {
+        if (type instanceof VarcharType) {
+            VarcharType varcharType = (VarcharType) type;
+            if (varcharType.getLength() <= MAX_IN_ROW_VARCHAR_LENGTH) {
+                return "varchar(" + varcharType.getLength() + ")";
+            }
+            else if (varcharType.getLength() <= 65535) {
+                return "text";
```

Comment:
What about "text"? After all, we're converting from Presto `varchar(x)` to MySQL `text`. 

From http://dev.mysql.com/doc/refman/5.7/en/storage-requirements.html, `text` needs "L + 2 bytes, where L < 2^16", and "L represents the actual length in bytes of a given string value.". Therefore, if we try to convert a `varchar(65536)` that contains non-ascii utf-8 characters, the data will exceed the storage requirements for `text`.